core/src/main/scala/com/raphtory/api/input/sources/CSVEdgeListSource.scala

```diff
@@ -0,0 +1,98 @@
+package com.raphtory.api.input.sources
+
+import com.raphtory.api.input.Graph.assignID
+import com.raphtory.api.input.{Graph, GraphBuilder, ImmutableProperty, Properties, Source, SourceInstance, Spout, SpoutInstance}
+import com.raphtory.internals.graph.GraphBuilderInstance
+import com.raphtory.spouts.{FileSpout, ResourceSpout}
+import com.raphtory.internals.time.DateTimeParser.{defaultParse => parseDateTime}
+
+import scala.language.implicitConversions
+
+class CSVEdgeListSource(override val spout: Spout[String], timeIndex: Int = 2, sourceIndex: Int = 0, targetIndex: Int = 1) extends Source {
+  override type MessageType = String
+
+  private var dateTimeFormat: Boolean = _
+  private var epochFormat: Boolean = _
+
+  private var longFormat: Boolean = _
+  private var stringFormat: Boolean = _
+
+
+  def buildCSVEdgeListGraph(graph: Graph, rawTime: String, source: String, target: String) = {
+    val timestamp = {
+      if (dateTimeFormat) {
+        parseDateTime(rawTime)
+      } else if (epochFormat) {
+        rawTime.toLong
+      } else {
+        throw new RuntimeException("Check timestamp is in Epoch or DateTime Format")
```

Comment:
Can we have something slightly more specific like s"$timestamp from $tuple did not conform to $dataTime format nor could it be interpreted as a epoch (long)"

Reply:
(Sorry just remembered we work this out with the first line don't we) may want to add that as part of the error i.e. "does not conform to what was seen on line 1"